src/main/java/bio/terra/cda/app/service/QueryService.java

```diff
@@ -110,23 +90,64 @@ protected JsonNode valueToJson(FieldValue value, Field field) {
     }
   }
 
-  private List<JsonNode> getJobResults(Job queryJob) {
+  public QueryResult getQueryResults(String queryId, int offset, int pageSize) {
+    final Job job = bigQuery.getJob(queryId);
+    if (job == null || !job.exists()) {
+      throw new RuntimeException("Unknown query " + queryId);
+    }
+    if (!job.isDone()) {
+      // If the Query is still running, return an empty result.
+      return new QueryResult(Collections.emptyList(), null);
+    }
+    return getJobResults(job, offset, pageSize);
+  }
+
+  public static class QueryResult {
+    public final List<Object> items;
+    public final Long totalRowCount;
+
+    QueryResult(List<JsonNode> items, Long totalRowCount) {
+      this.items = new ArrayList<>(items);
```

Comment:
Why do you need to wrap this?

Reply:
Same as before `List<X>` -> `List<Object>` conversion